sig-scalability/charter.md

```diff
@@ -0,0 +1,110 @@
+- Ensuring that each official Kubernetes release satisfies all scalability and
+performance related requirements, as stated in "Kubernetes scalability" definition.
+- Establishing and documenting best practises on how to design and/or implement
+Kubernetes features in scalable and performant way. Educating contributors and
+consulting individual designs/implementations to ensure that those are widely used.
+Example artifacts:
+  - [Scalability governance](https://github.com/kubernetes/community/blob/master/sig-scalability/governance)
+
+
+TODO: Figure out how to phrase finding bottlenecks, coordinating improvements
+and architectural changes, etc.
+
+### Out of scope
+
+- Improving performance/scalability of features falling into charters of
+individual SIGs.
+
+
+## What can we do/require from other SIGs
+Scalability and performance are horizontal aspects of the system - changes in a
+single place of Kubernetes may affect the whole system. As a result, to
+effectively ensure Kubernetes scales, we need a special cross-SIG privileges.
+
+- We can rollback any merged PR if it has been identified as a cause of any
+  [performance/scalability SLOs] regression. The offending PR should only be
+  merged again after proving to pass  tests at scale.
+- We can pause the merge queue in case of a regression observed until a particular
+  PR has been identified as cause of the regression and regression has been
+  mitigated. The “Rules of engagement” of pausing merge-queue and rationale for
+  necessity of its introduce are explained in a separate doc. <br/>
+  TODO(wojtek-t, shyamjvs): Write it down and link here.
+- We require significant changes (in terms of impact, such as: update of etcd,
+  update of Go version, major architectural changes, etc.) may only be merged:
+  - with an explicit approval from a [SIG-scalability approver](#sig-scalability-approvers)
+    and
+  - after having passed performance testing on biggest supported clusters (unless
+    found unnecessary by scalability approver)
+- We can block a feature from transitioning to Beta status if (when turned on) it
+  causes a significant degradation of overall Kubernetes scalability/performance.
+  (Ideally it would be “SLI degradation of more than X%” or “breaking SLO”, but
+  initially it may also be SIG-scalability decision based on public test results).
+- We can block a feature from transitioning to GA status if it cannot be used at
+  scale.
+- We can require a SIG to introduce a regression-catching benchmark test for a
+  scalability-critical functionality.
+
+For the record, by regression above we mean a regression identified by the set 
+of release-blocking scalability/performance tests (as defined by
+sig-release-master-blocking group of test suites).
+
+[performance/scalability SLOs]: https://github.com/kubernetes/community/blob/master/sig-scalability/slos/slos.md
+
+### SIG Scalability approvers
```

Comment:
this shouldn't be here.  
this should be in the sigs.yaml

Reply:
So I thought a bit more about that, and:
- longer term it may make sense to move to sigs.yaml
- currently, it's not supported there (it's somewhat sig-specific role) and i don't want to add that now.

Reply:
Yes it is, please see other sigs and remove.

Reply:
Can you please point me to one example? I claim that anything that exists is not what we want (i.e. this list may contain people that aren't even scalability sig members, but they have wide system knowledge and can access scalability impact of changes).

Reply:
We list the "chairs" and "leads" in the main sigs.yaml, approvers is a concept in the OWNERs which is subject to change over time.  Eventually, folks want to get to a canonical location for the OWNERs, but the charter is not the appropriate location for what is essentially "OWNERs data", and has been strip'd out from other charters.

Reply:
I gave up for now and switched to tech leads with a TODO to create a dedicated group for it.